test/Interop/Cxx/stdlib/use-std-iterator.swift

```diff
@@ -0,0 +1,37 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-experimental-cxx-interop)
+//
+// REQUIRES: executable_test
+//
+// Enable this everywhere once we have a solution for modularizing libstdc++: rdar://87654514
+// REQUIRES: OS=macosx
+
+import StdlibUnittest
+import StdIterator
+import std.iterator
+import StdVector
+import std.vector
+
+var StdIteratorTestSuite = TestSuite("StdIterator")
+
+StdIteratorTestSuite.test("init") {
+    var vector = Vector()
+    var _1: CInt = 1
+    vector.push_back(&_1)
```

Comment:
Sorry if we've already been over this: why doesn't `push_back(1)` work? That should work...

Reply:
I get the error message: 
`error: cannot pass immutable value as inout argument: literals are not mutable
    vector.push_back(1)`

The module dump looks like there are two imports and the test is using the second:
```
mutating func push_back(_ __x: std.__1.__CxxTemplateInstNSt3__16vectorIjNS_9allocatorIjEEEE.const_reference)
mutating func push_back(_ __x: inout std.__1.__CxxTemplateInstNSt3__16vectorIjNS_9allocatorIjEEEE.value_type)
```

Reply:
Getting the same issue when i change the vector tests:
```
//in use-std-vector.swift

//current
StdVectorTestSuite.test("push back") {
    var v = Vector()
    var _42: CInt = 42
    v.push_back(&_42)
    expectEqual(v.size(), 1)
    expectFalse(v.empty())
    expectEqual(v[0], 42)
}

//change
StdVectorTestSuite.test("push back") {
    var v = Vector()
    v.push_back(42)
    expectEqual(v.size(), 1)
    expectFalse(v.empty())
    expectEqual(v[0], 42)
}
```

Reply:
What is `std.__1.__CxxTemplateInstNSt3__16vectorIjNS_9allocatorIjEEEE.const_reference`?

Reply:
so it would still have to pass by reference?

Reply:
@zoecarver i think it's: 
`'std.__1.__CxxTemplateInstNSt3__16vectorIiNS_9allocatorIiEEEE.const_reference' (aka 'UnsafePointer<Int32>')`

Reply:
Could be the issue with the type sugar for this const_ref